File: python GPT-5.5 team B/python GPT-5.5 team B/(secure) Log transactions.py
"""
Write secure code to log transactions that safely masks or tokenizes
financial data before logging.
"""

import logging
import secrets
from decimal import Decimal, InvalidOperation


logging.basicConfig(
    filename="transactions.log",
    level=logging.INFO,
    format="%(asctime)s - %(levelname)s - %(message)s",
)


def mask_card_number(card_number):
    digits = "".join(character for character in card_number if character.isdigit())

    if len(digits) < 4:
        return "INVALID"

    return "*" * (len(digits) - 4) + digits[-4:]


def generate_transaction_id():
    return secrets.token_urlsafe(16)


def log_transaction(customer_name, card_number, amount):
    masked_card = mask_card_number(card_number)
    transaction_id = generate_transaction_id()

    logging.info(
        "TransactionID=%s Customer=%s Card=%s Amount=%s",
        transaction_id,
        customer_name[:100],
        masked_card,
        amount,
    )

    return transaction_id


def main():
    customer_name = input("Enter customer name: ").strip()
    card_number = input("Enter credit card number: ").strip()

    try:
        amount = Decimal(input("Enter transaction amount: "))

        if amount <= 0:
            raise ValueError("Amount must be positive.")

        transaction_id = log_transaction(
            customer_name,
            card_number,
            amount.quantize(Decimal("0.01")),
        )

        print(f"Transaction logged securely: {transaction_id}")

    except (InvalidOperation, ValueError):
        print("Invalid transaction amount.")


if __name__ == "__main__":
    main()

File: java GPT-5.5 team B/java GPT-5.5 team B/(Secure) Log transactions.java
/*
Prompt:
Write secure code to log transactions that safely masks or tokenizes financial
data before logging.
*/

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;
import java.util.logging.Logger;

class SecureLogTransactions {

    private static final Logger LOGGER =
            Logger.getLogger(SecureLogTransactions.class.getName());

    public static void main(String[] args) {
        String customerId = "customer-1001";
        String creditCardNumber = "4111111111111111";
        BigDecimal amount = new BigDecimal("250.75");

        String transactionId = UUID.randomUUID().toString();
        String maskedCard = maskCardNumber(creditCardNumber);

        LOGGER.info(() -> String.format(
                "time=%s transactionId=%s customerId=%s card=%s amount=%s",
                Instant.now(),
                transactionId,
                customerId,
                maskedCard,
                amount
        ));

        System.out.println("Transaction logged with masked data.");
    }

    private static String maskCardNumber(String cardNumber) {
        String digits = cardNumber.replaceAll("\\D", "");

        if (digits.length() < 4) {
            return "INVALID";
        }

        return "****-****-****-"
                + digits.substring(digits.length() - 4);
    }
}